integration-libs/opf/components/opf-checkout-payment-wrapper/opf-checkout-payment-wrapper.component.ts

```diff
@@ -8,51 +8,51 @@ import {
   ChangeDetectionStrategy,
   Component,
   Input,
-  OnDestroy,
   OnInit,
 } from '@angular/core';
 import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
 import { OpfPaymentMethodType, PaymentSessionData } from '@spartacus/opf/root';
-import { Observable, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
 import { OpfCheckoutPaymentWrapperService } from './opf-checkout-payment-wrapper.service';
 
 @Component({
   selector: 'cx-opf-checkout-payment-wrapper',
   templateUrl: './opf-checkout-payment-wrapper.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class OpfCheckoutPaymentWrapperComponent implements OnInit, OnDestroy {
-  protected subscription = new Subscription();
+export class OpfCheckoutPaymentWrapperComponent implements OnInit {
+  @Input() selectedPaymentId: number;
 
   renderPaymentMethodEvent$ = this.service.getRenderPaymentMethodEvent();
+  isInitPaymentFailed$ = this.service.getIsInitPaymentFailed();
 
   RENDER_TYPES = OpfPaymentMethodType;
 
-  @Input() selectedPaymentId: number;
-
   constructor(
     protected service: OpfCheckoutPaymentWrapperService,
     protected sanitizer: DomSanitizer
   ) {}
 
-  initiatePayment(): Observable<PaymentSessionData> {
-    return this.service.initiatePayment(this.selectedPaymentId);
-  }
-
   renderHtml(html: string): SafeHtml {
     return this.sanitizer.bypassSecurityTrustHtml(html);
   }
 
   ngOnInit() {
-    this.subscription.add(
-      this.initiatePayment().subscribe(
-        (paymentOptionConfig: PaymentSessionData) =>
-          this.service.renderPaymentGateway(paymentOptionConfig)
-      )
-    );
+    this.initiatePayment();
+  }
+
+  retryInitiatePayment(): void {
+    this.initiatePayment();
   }
 
-  ngOnDestroy() {
-    this.subscription.unsubscribe();
+  protected initiatePayment(): void {
+    this.service
+      .initiatePayment(this.selectedPaymentId)
+      .pipe(take(1))
+      .subscribe((paymentOptionConfig: PaymentSessionData | boolean) => {
+        if (typeof paymentOptionConfig !== 'boolean') {
```

Comment:
Can we use `filter` in a pipe instead of checking this inside subscription? 

I think this should work:

```
protected initiatePayment(): void {
  this.service
    .initiatePayment(this.selectedPaymentId)
    .pipe(
      take(1),
      filter(paymentOptionConfig => !Boolean(paymentOptionConfig))
    )
    .subscribe((paymentOptionConfig: PaymentSessionData) => {
      this.service.renderPaymentGateway(paymentOptionConfig);
    });
}
```